### PR TITLE
Optimize utilization card by splitting items to smaller components ea…

### PR DESCRIPTION
**Pull request by rawagner**
…ch fetching its own data

### CODE REVIEW

File: frontend/public/components/dashboard/dashboards-page/overview-dashboard/utilization-card.tsx [hunk 1]
```diff
@@ -117,135 +120,112 @@ const getQueries = (flags: FlagsObject) => {
...
+export const UtilizationCard = connectToFlags(
+  ...plugins.registry.getRequiredFlags([isDashboardsOverviewUtilizationItem]),
+)(({ flags }) => {
+  const queries = React.useMemo(() => getQueries(flags), [flags]);
+  const [timestamps, setTimestamps] = React.useState<Date[]>();
+  const [duration, setDuration] = React.useState(metricDurations[0]);
 
   const cpuPopover = React.useCallback(
-    ({ current }) => (
+    React.memo<TopConsumerPopoverProp>(({ current }) => (
```
Comment (mareklibra) [inline frontend/public/components/dashboard/dashboards-page/overview-dashboard/utilization-card.tsx]: Again. Just for readability - move the `memo()` call up and use well-named variable for it, please.

File: frontend/public/components/dashboard/dashboards-page/overview-dashboard/utilization-card.tsx [hunk 2]
```diff
@@ -117,135 +120,112 @@ const getQueries = (flags: FlagsObject) => {
...
         position={PopoverPosition.top}
       />
-    ),
+    )),
     [],
   );
 
   const memPopover = React.useCallback(
-    ({ current }) => (
+    React.memo<TopConsumerPopoverProp>(({ current }) => (
```
Comment (mareklibra) [inline frontend/public/components/dashboard/dashboards-page/overview-dashboard/utilization-card.tsx]: Likewise

File: frontend/public/components/dashboard/project-dashboard/utilization-card.tsx [hunk 3]
```diff
@@ -4,225 +4,168 @@ import DashboardCard from '@console/shared/src/components/dashboard/dashboard-ca
...
-              query: getTopConsumerQueries(projectName)[ProjectQueries.PODS_BY_NETWORK],
-              model: PodModel,
-              metric: 'pod',
-            },
-          ]}
-          humanize={humanizeDecimalBytesPerSec}
-          namespace={projectName}
-          position={PopoverPosition.top}
+        <PrometheusUtilizationItem
+          title="Network Transfer in/out"
```
Comment (mareklibra) [inline frontend/public/components/dashboard/project-dashboard/utilization-card.tsx]: `IN/OUT` instead?

Comment (rawagner) [reply frontend/public/components/dashboard/project-dashboard/utilization-card.tsx]: we should have just a `Network transfer` here, so I changed it accordingly.

File: frontend/packages/metal3-plugin/src/components/baremetal-hosts/dashboard/UtilizationCard.tsx [hunk 4]
```diff
@@ -210,72 +124,63 @@ const UtilizationCard: React.FC<UtilizationCardProps> = ({
...
-        <UtilizationItem
+        <PrometheusUtilizationItem
           title="Number of pods"
-          data={numberOfPodsStats}
-          error={numberOfPodsError}
-          isLoading={!numberOfPods}
-          humanizeValue={(v) => ({ string: `${v}`, value: v as number, unit: '' })}
-          query={queries[HostQuery.NUMBER_OF_PODS].utilization}
+          utilizationQuery={queries[HostQuery.NUMBER_OF_PODS].utilization}
+          humanizeValue={React.useCallback(
```
[the remark this answers is not in the document]
Comment (rawagner) [reply frontend/packages/metal3-plugin/src/components/baremetal-hosts/dashboard/UtilizationCard.tsx]: fixed

**Pull request merged**